src/ucs/arch/x86_64/cpu.c

```diff
@@ -333,4 +469,96 @@ void ucs_cpu_init()
 #endif
 }
 
+void ucs_arch_get_cache_size(ucs_cache_size_t *cache, size_t length)
+{
+    x86_cache_line_reg_info_t cache_info;
+    x86_cache_line_reg_info_t line_info;
+    x86_cpu_registers reg;
+    uint32_t sets;
+    uint32_t i, t, r, l4;
+    uint32_t max_iter;
+    size_t c;
+    int level1_only; /* level 1 cache only supported */
+
+    ucs_assert(length <= ucs_static_array_size(x86_cpu_cache));
+
+    /* Get CPU ID and vendor - it will reset cache iteration sequence */
+    if (ucs_arch_get_cpu_vendor() != UCS_CPU_VENDOR_INTEL) {
+        return;
+    }
+
+    ucs_x86_cpuid(X86_CPUID_GET_BASE_VALUE,
+                  ucs_unaligned_ptr(&reg.eax), ucs_unaligned_ptr(&reg.ebx),
+                  ucs_unaligned_ptr(&reg.ecx), ucs_unaligned_ptr(&reg.edx));
+    if (reg.eax < X86_CPUID_GET_CACHE_INFO) {
+        return;
+    }
+
+    level1_only = 0;
+
+    for (i = 0, max_iter = 1; i < max_iter; i++) {
+        ucs_x86_cpuid(X86_CPUID_GET_CACHE_INFO,
+                      ucs_unaligned_ptr(&reg.eax), ucs_unaligned_ptr(&reg.ebx),
+                      ucs_unaligned_ptr(&reg.ecx), ucs_unaligned_ptr(&reg.edx));
+
+        if (i == 0) { /* on first iteration get max iteration number */
+            max_iter     = reg.max_iter;
+            reg.max_iter = 0; /* mask iteration register from processing */
+        }
+
+        for (r = 0; r < ucs_countof(reg.reg); r++) {
+            if (ucs_test_all_flags(reg.reg[r].value, X86_CPU_CACHE_RESERVED)) {
+                continue;
+            }
+
+            for (t = 0; (t < ucs_countof(reg.reg[r].tag)) && (reg.reg[r].tag[t] != 0); t++) {
+                switch(reg.reg[r].tag[t]) {
+                case X86_CPU_CACHE_TAG_L1_ONLY:
+                    level1_only = 1;
+                    break;
+                case X86_CPU_CACHE_TAG_LEAF4:
+                    for (l4 = 0;; l4++) {
+                        ucs_x86_cpuid_ecx(X86_CPUID_GET_LEAF4_INFO, l4,
+                                          ucs_unaligned_ptr(&cache_info.reg), ucs_unaligned_ptr(&line_info.reg),
```

Comment:
line too long?

Reply:
on VSCode it doesn't looks like too long :)
fixed